website/hw/hw4.rst

```diff
@@ -180,103 +218,517 @@ As you are stepping through your program, you might want to be able to
+
+The editor won't actually start just yet (this is normal!) and you'll see something
+like this::
+
+    Process build/micro created; pid = 549651
+    Listening on port 50000
+
+.. note::
+
+   If you see the following message::
+
+        Can't bind address: Address already in use.
+        Exiting
+
+   Then pick a number other than 50000 in the parameter to ``gdbserver``
+   (any number between 50000 and 60000 should work). This number is known
+   as a "port", and we will be using it to connect to the editor from GDB.
+   We have to pick a number that is unique within the machine we are running
+   on; this means that if you are running this on one of the CS Linux server,
+   and multiple people choose 50000, only one of them will actually be able
+   to run the debugger.
+
+Now, on the second terminal, run this::
+
+    gdb build/micro
+
+This will open the usual GDB prompt but, instead of typing "run", we need to
+connect to the editor that is running on the other terminal. You need
+to run the following command from the GDB prompt::
+
+    target remote :50000
+
+You will see something like this::
+
+    (gdb) target remote :50000
+    Remote debugging using :50000
+    Reading /lib64/ld-linux-x86-64.so.2 from remote target...
+    warning: File transfers from remote targets can be slow. Use "set sysroot" to access files locally instead.
+    Reading /lib64/ld-linux-x86-64.so.2 from remote target...
+    Reading symbols from target:/lib64/ld-linux-x86-64.so.2...
+    Reading /lib64/ld-2.31.so from remote target...
+    Reading /lib64/.debug/ld-2.31.so from remote target...
+    Reading /usr/lib/debug//lib64/ld-2.31.so from remote target...
+    Reading /usr/lib/debug/lib64//ld-2.31.so from remote target...
+    Reading target:/usr/lib/debug/lib64//ld-2.31.so from remote target...
+    (No debugging symbols found in target:/lib64/ld-linux-x86-64.so.2)
+    0x00007ffff7fd0100 in ?? () from target:/lib64/ld-linux-x86-64.so.2
+
+You can ignore the warning about file transfers being slow, or the message about
+"No debugging symbols" (this refers to a system library, not to our editor)
+
+You can now run any of the commands we've seen previously, with one caveat: once
+you want to start running the editor, you must use the ``continue`` command,
+and not the ``run`` command. In fact, try just running the ``continue`` command;
+GDB will show something like this::
+
+    (gdb) continue
+    Continuing.
+    Reading /lib/x86_64-linux-gnu/libc.so.6 from remote target...
+    Reading /lib/x86_64-linux-gnu/libc-2.31.so from remote target...
+    Reading /lib/x86_64-linux-gnu/.debug/libc-2.31.so from remote target...
+    Reading /usr/lib/debug//lib/x86_64-linux-gnu/libc-2.31.so from remote target...
+    Reading /usr/lib/debug//lib/x86_64-linux-gnu/libc-2.31.so from remote target...
+
+And, on the first terminal, the editor will start running.
+
+If you need to quit the editor, we suggest doing so from the first terminal
```

Comment:
May be helpful to note that if you quit the editor, you can still re-link to the editor session from gdb just by doing `target remote :50000` again (you don't need to quit gdb every time)

Reply:
Good point, I've added instructions for that.